### ISSUE TITLE
Expected 406 on Accept: application/json, got 200 OK

### ISSUE THREAD

**Issue by preliot** [laravel-json-api/laravel#184]
I'm using postman to test our new API. 

Noticed that when i set `Accept: application/json`, the API still works. Every other Accept header, not being `application/vnd.api+json`, will return a 406 as expected. This should not happen, see [https://laraveljsonapi.io/docs/2.0/requests/compliance.html#accept-header](https://laraveljsonapi.io/docs/2.0/requests/compliance.html#accept-header).

This is correct:

**LaravelJsonApi\Laravel\Http\Requests\ResourceQuery.php**

```php
/**
     * @var string[]
     */
    protected array $mediaTypes = [
        self::JSON_API_MEDIA_TYPE,
    ];
```
This seems to produce the incorrect result:

```php
/**
     * @return bool
     */
    protected function isAcceptableMediaType(): bool
    {
        return $this->accepts($this->mediaTypes());
    }
```

`Illuminate\Http\Concerns\InteractsWithContentTypes.php` > `matchesType()` assumes `application/json` and `application/vnd.api+json` are identical.

**Comment by preliot** [comment]
- laravel-json-api/laravel: 2.3.0
- laravel/framework: 8.83.8

**Comment by lindyhopchris** [comment]
Hi! Thanks for reporting this. I'll definitely look into it as you're right that it looks like a bug.

**Comment by lindyhopchris** [comment]
Fix will be in `2.4.0`
